src/components/tabs/tab.styles.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { css } from '@emotion/react';
+import { UseEuiTheme } from '../../services';
+import { euiTitle } from '../title/title.styles';
+
+export const euiTabStyles = ({ euiTheme }: UseEuiTheme) => {
+  return {
+    euiTab: css`
+      display: flex;
+      cursor: pointer;
+      flex-direction: row;
+      align-items: center;
+      font-weight: ${euiTheme.font.weight.semiBold};
+      gap: ${euiTheme.size.s};
+
+      &:focus {
+        background-color: transparent;
+        outline-offset: -${euiTheme.focus.width};
+      }
+    `,
+    // sizes
+    s: css`
+      padding: 0 ${euiTheme.size.xs};
+    `,
+    m: css`
+      padding: 0 ${euiTheme.size.xs};
+    `,
+    l: css`
+      padding: 0 ${euiTheme.size.xs};
+    `,
+    xl: css`
+      padding: ${euiTheme.size.s} ${euiTheme.size.xs};
+    `,
+    // variations
+    expanded: css`
+      flex-basis: 0%;
+      flex-grow: 1;
+      justify-content: center;
+    `,
+    selected: css`
+      box-shadow: inset 0 calc(${euiTheme.border.width.thick} * -1) 0
+        ${euiTheme.colors.primary};
+    `,
+    disabled: css`
+      cursor: not-allowed;
+
+      .euiTab {
+        box-shadow: inset 0 calc(${euiTheme.border.width.thin} * -1) 0
+          ${euiTheme.colors.disabledText};
+      }
+
+      .euiTab__prepend,
+      .euiTab__append {
+        color: ${euiTheme.colors.disabledText};
+      }
+
+      .euiTab.euiTab__isSelected {
+        box-shadow: inset 0 calc(${euiTheme.border.width.thick} * -1) 0
+          ${euiTheme.colors.disabledText};
+      }
```

Comment:
Is this missing the `&` to work? 

```suggestion
      &.euiTab__isSelected {
        box-shadow: inset 0 calc(${euiTheme.border.width.thick} * -1) 0
          ${euiTheme.colors.disabledText};
      }
```

Reply:
It is possible for a tab to be selected and disabled (see below, first tab):

<img width="549" alt="" src="https://user-images.githubusercontent.com/549407/197636790-61cccc9b-3e86-49b6-805c-7e7d3fa177ea.png">

My vote is to leave this CSS in but rewrite it to remove nesting and use more conditional JS logic.

Reply:
@daveyholler LMK if you'd rather me simply push up my code change requests to this PR as well - happy to do so as I'm running low on Wording Good today and it might be easier for me to just code up what I'm looking for